experimenter/experimenter/experiments/migrations/0303_backfill_qa_run_date_from_changelog.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 5.2.2 on 2025-01-20 15:50
+
+from django.db import migrations
+
+
+def backfill_qa_run_date_from_changelog(apps, schema_editor):
+    NimbusExperiment = apps.get_model("experiments", "NimbusExperiment")
+    NimbusChangeLog = apps.get_model("experiments", "NimbusChangeLog")
+
+    experiments_to_update = NimbusExperiment.objects.filter(
+        qa_run_date__isnull=True
+    ).exclude(
+        qa_status="NOT_SET"
+    )
+
+    for experiment in experiments_to_update:
+        # Get changelogs for this experiment ordered by most recent first
+        changelogs = list(
+            NimbusChangeLog.objects.filter(experiment=experiment)
+            .order_by("-changed_on")
+            .values("experiment_data", "changed_on")
+        )
```

Comment:
Why don't we do something like

```py
experiment.changes.all().exclude(experiment_data__qa_status='NOT SET').order_by('changed_on').first().changed_on
```

Just get the first one where it goes from NOT SET to something?  Faster to do in the db than in memory.